api-core/src/main/java/com/vimeo/networking2/internal/ErrorHandlingCallAdapter.kt

```diff
@@ -15,7 +15,7 @@ import java.util.concurrent.Executor
  * @param callbackExecutor          The executor upon which the caller is notified.
  * @param responseBodyConverter     Converter to parse the error response into a [ApiError].
  */
-internal class ErrorHandlingCallAdapter<T>(
+internal class ErrorHandlingCallAdapter<T : Any>(
```

Comment:
Just curious but what was the lint warning to add this? `Any` doesn't seem necessary to me here.

Reply:
The upper bound on `T` in `requireNotNull` is `T : Any`, so I can't pass a `T` to it without getting a build error. Since we never expect a nullable type back as a result of a call (null is an error), we can safely set the upper bound to `T`.